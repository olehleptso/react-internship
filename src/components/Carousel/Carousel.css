.carousel {
    margin: 0 auto;
    height: 100vh;
}

.carousel-inner {
    display: flex;
    overflow: hidden;
}

.carousel-item-block{
    display: flex;
    min-width: 100%;
    width: 100%;
    align-items: center;
    justify-content: center;
    transition: 1s cubic-bezier(0.39, 0.575, 0.565, 1);
}