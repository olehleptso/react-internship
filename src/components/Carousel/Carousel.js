import { useState, useEffect } from 'react';
import CarouselControls from '../CarouselControls/CarouselControls';
import CarouselItem from '../CarouselItem/CarouselItem';
import ModalSlider from '../ModalSlider/ModalSlider';
import './Carousel.css';

function Carousel () {

    const [currentSlide, setCurrentSlide] = useState(0);
    const [projectData, setProjectData] = useState([]);
    const [openModal, setOpenModal] = useState(false);

    const prev = () => {
        const index = currentSlide > 0 ? currentSlide - 1 : projectData.length - 1;
        setCurrentSlide(index);
      }
      
    const next = () => {
        const index = currentSlide < projectData.length - 1 ? currentSlide + 1 : 0;
        setCurrentSlide(index);
      }
    
    const more = () => {
        setOpenModal(true);
    }
    
    useEffect(()=>{
        async function getProjectData() {
            const response = await fetch('http://localhost:4000/projects');
            const res = await response.json();
            setProjectData(res);
        }
        getProjectData();
    }, []);
    
    return (
        <div data-testid='carousel'>
            <div className='carousel' id="projects">
                <div className='carousel-inner'>
                    {projectData.map((data, index) => {
                        return <div 
                                    className='carousel-item-block' key={index}
                                    style={{ transform: `translate(-${currentSlide * 100}%)`}}
                                >
                                    <CarouselItem data={data}/>
                                </div>
                    })}
                </div>
                <CarouselControls prev={prev} next={next} more={more}/>
            </div>
            {openModal && <ModalSlider closeModal={setOpenModal} data={projectData[currentSlide].images} />}
        </div>
    )
}

export default Carousel;