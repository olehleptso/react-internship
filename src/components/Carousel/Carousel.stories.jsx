import React from 'react';
import Carousel from './Carousel';

export default {
  title: 'Carousel',
  component: Carousel,
};

const Template = (args) => <Carousel {...args} />;

export const Standart = Template.bind({});
