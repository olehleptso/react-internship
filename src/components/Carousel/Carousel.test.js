import { render, screen } from '@testing-library/react';
import Carousel from './Carousel';


test('renders Carousel component', () => {
    render(<Carousel />);
    const CarouselSec = screen.getByTestId('carousel');
    expect(CarouselSec).toBeInTheDocument();
});