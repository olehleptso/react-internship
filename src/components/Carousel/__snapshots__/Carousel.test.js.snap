// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders search Carousel component 1`] = `
<DocumentFragment>
  <div
    data-testid="carousel"
  >
    <div
      class="carousel"
      id="projects"
    >
      <div
        class="carousel-inner"
      />
      <div
        class="carousel-controls"
      >
        <div
          class="controls-inner"
        >
          <button
            class="previous btn"
          >
            &lt; previous project
          </button>
          <img
            src="squares.png"
          />
          <button
            class="next btn"
          >
            next project &gt;
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
